pgrx-pg-sys/src/submodules/elog.rs

```diff
@@ -103,6 +103,36 @@ impl From<i32> for PgLogLevel {
 ///
 /// The output these logs goes to the PostgreSQL log file at `DEBUG5` level, depending on how the
 /// [PostgreSQL settings](https://www.postgresql.org/docs/current/runtime-config-logging.html) are configured.
+///
+/// ## Adding Details and Hints
+///
+/// Additionally, you can use specify a `detail` and/or `hint` to be included with the report.
+/// After the string literal to be used as the message and any formatting arguments, add a semicolon.
+/// Then assign a string literal to `detail` or `hint`, optionally followed by a list of
+/// comma separated formatting arguments. The `detail` and `hint` options can be used together or
+/// separately but `detail` must come before `hint` and both must end with a semicolon.
+///
+/// ## Examples
+///
+/// ```rust,no_run
+/// // just like `println!` and `format!`
+/// debug5!("a simple message");
+/// debug5!("or a formatted message: {:?}", "pgrx rocks!");
+/// // include details and hints
+/// debug5! {
+///     "add details if you want";
+///     detail = "...";
+/// }
+/// debug5! {
+///     "or a message with just a hint";
+///     hint = "...";
+/// }
+/// debug5! {
+///     "put it all together for {} {}", "great", "success!";
+///     detail = "extra {}", "info";
+///     hint = "{} helpful", "very";
+/// }
+/// ```
```

Comment:
I recommend `cargo test --doc` to make sure these examples work.

Reply:
Thanks for the tip! There were a couple issues:
- I needed to include `use pgrx_pg_sys::xxx;` in each example
- The compiler was struggling to identify the string literal in the `format!` calls. Passing the components through `format_args!` seems to have made things clearer.

While the implementation is considerably more verbose than what I originally anticipated, the end result delivers the ergonomics I was aiming for.